### ISSUE TITLE
Is it possible to set a mark and jump to it later like in `less`?

### ISSUE THREAD

**Issue by your-diary** [walles/moar#175]
In `less` (and `vim`), a mark can be set on the current line and you can jump to it later.

1. Set a mark via <kbd>'</kbd><kbd>a</kbd> (a single quote followed by any single character).

2. Jump to it via <kbd>m</kbd><kbd>a</kbd> (an em followed by the mark character).

It seems this doesn't work in `moar`.

---

By the way, I created this issue as `README.md` says

> If you come from Less and find Moar confusing or hard to migrate to, [please report it](https://github.com/walles/moar/issues)!

**Comment by walles** [comment]
Thanks for reporting this!

Short answer: not right now

In what situations do you use marks? I never did so I'm trying to understand how one might want to use them.

**Comment by your-diary** [comment]
Marks can be used for any purposes when you want to mark a line and jump to it later.

*One* example is that you execute `PAGER=less man xyz` when you trying to learn `xyz` command *for the first time*.
You may want to first read through the entire documentation while setting marks and temporarily skip too detailed sections.
After you hit the end of the document, you now have shallow but wide understanding for the command, so it's time to jump back to the skipped detailed sections.
Using marks, you don't have to memorize which sections you have skipped; you just type <kbd>m</kbd><kbd>a</kbd>, <kbd>m</kbd><kbd>b</kbd>, etc.

*Another* example is when you code review a pull request via `git log -p develop..HEAD --reverse`.
This command lists the commits with their diffs in the current branch in old to new order.
In other words, you can first see the oldest commit and scroll down to see newer commits.
Similar to the former example, you may sometimes want to temporarily skip a large commit or a single file in a commit.
You can then just set a mark to it and jump to it later.

These examples are actually derived from my daily use-cases.
Especially, I code review PRs ***almost everyday*** while setting marks.

Further reading: [*vim how do you use marks - Google Search*](https://www.google.com/search?q=vim+how+do+you+use+marks)

**Comment by postsolar** [comment]
I second marks for browsing man pages. Often, different sections would cross-reference each other, so each time I want to temporarily open another section I would need to perform a lot of scrolling and/or searching to get back where I was.

**Comment by walles** [comment]
I just tried this with `less`, and it's <kbd>m</kbd> to set marks and <kbd>'</kbd> to visit them.

**Comment by walles** [comment]
This is out now, please try it!
https://github.com/walles/moar/releases/tag/v1.22.0

Since I'm not using this myself I'd very much appreciate feedback, good or bad.

Right now in the go-to-mark prompt I'm listing the marks in alphabetic order, maybe this should be in line number order instead? Or would that be confusing? Patches / opinions welcome.

**Comment by your-diary** [comment]
@walles 
Thank you very much!
It works well.

> Right now in the go-to-mark prompt I'm listing the marks in alphabetic order, maybe this should be in line number order instead? Or would that be confusing?

I have no opinion about that because I don't look at the prompt after pressing <kbd>'</kbd>. (Normally I memorize which marks I have set.)
At least, the current display is NOT confusing or annoying. I like it.

**Comment by postsolar** [comment]
@walles 
Thank you for adding this feature, it works well, apparently even with non-ASCII characters.
Personally I don't have a preference for the order of marks in the marks prompt.
